wire-ios/Wire-iOS/Sources/UserInterface/Settings/Backup/BackupLocalStore/BackupLocalStore.swift

```diff
@@ -0,0 +1,170 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireBackup
+import WireDataModel
+import WireDomain
+import WireFoundation
+
+struct BackupLocalStore<
+    UserLocalStore, ConversationLocalStore, MessageLocalStore
+>: BackupLocalStoreProtocol, @unchecked Sendable where
+    UserLocalStore: UserLocalStoreProtocol,
+    MessageLocalStore: MessageLocalStoreProtocol,
+    ConversationLocalStore: ConversationLocalStoreProtocol {
+
+    /// The context to call `perform(schedule:_:)` on if needed.
+    private let context: NSManagedObjectContext
+
+    private let userLocalStore: UserLocalStore
+    private let conversationLocalStore: ConversationLocalStore
+    private let messageLocalStore: MessageLocalStore
+
+    func countModels() async throws -> (userCount: Int, conversationCount: Int, messageCount: Int) {
+        let userCount = try await userLocalStore.totalUserCountForBackup()
+        let conversationCount = try await conversationLocalStore.totalConversationCountForBackup()
+        let messageCount = try await messageLocalStore.totalMessageCountForBackup()
```

Comment:
suggestion: rather than wrapping these stores, I suggest to take their implementations (which were added for the backup feature) and have them here in a single `await context.perform { ... }` block so we don't switch back and forth unnecessarily.

Reply:
done 2ff3ec2c48e494db7de5418f00d22f4e43dd8447